libp2p/protocols/pubsub/pubsubpeer.nim

```diff
@@ -248,19 +292,53 @@ proc sendEncoded*(p: PubSubPeer, msg: seq[byte]) {.async.} =
     debug "No send connection", p, msg = shortLog(msg)
     return
 
-  trace "sending encoded msgs to peer", conn, encoded = shortLog(msg)
+  trace "sending encoded msg to peer", conn, encoded = shortLog(msg)
+  await sendMsgContinue(conn, conn.writeLp(msg))
```

Comment:
should we use this same pattern here?
```

    let f = conn.writeLp(msg)
    if not f.completed():
      sendMsgContinue(conn, f)
    else:
      f
```

we could also move this if/else inside `sendMsgContinue`.

Reply:
the reason for doing the f.completed check is to avoid an allocation that happens inside `sendMsgContinue` - the effect here would be negligible because only a small subset of writes actually go to the slow version (I hesitated doing it for the hot path even - it doesn't _greatly_ matter but on the other hand, there's a lot of sendMsg calls).

Reply:
> move this if/else inside

inside, it doesn't matter any more - the code generated by chronos is reasonably efficient once you get past the initial bump of calling an `{.async.}` function